src/v/cluster/panda_link_table.h

```diff
@@ -36,6 +36,25 @@ class panda_link_table {
     /// Restores a panda link table from a snapshot
     void reset_links(map_t);
 
+    /// Finds link by name
+    std::optional<std::reference_wrapper<const model::panda_link_metadata>>
+    find_link_by_name(std::string_view name) const;
+    std::optional<std::reference_wrapper<const model::panda_link_metadata>>
+    find_link_by_name(const model::panda_link_name& name) const;
+    /// Finds link by id
+    std::optional<std::reference_wrapper<const model::panda_link_metadata>>
+    find_link_by_id(model::panda_link_id id) const;
+    /// Finds link ID by name
+    std::optional<model::panda_link_id>
+    find_id_by_name(std::string_view name) const;
+    std::optional<model::panda_link_id>
+    find_id_by_name(const model::panda_link_name& name) const;
+
+    /// Upserts a link, if the ID classes, throws a std::logic_error
+    void upsert_link(model::panda_link_id, model::panda_link_metadata);
+    /// Removes a link by ID
+    void remove_link(const model::panda_link_name&);
```

Comment:
Those two methods should be private, the only way link table can be updated should be by applying the command